### PR TITLE
feat: add video support for livekit video track

### PR DESCRIPTION
**Pull request by pbosio**
## What does this PR change?
*adds api to fetch active livekit video streams using sdk
```
import { getActiveVideoStreams } from "~system/CommsApi"
executeTask(async () => {
    const result = await getActiveVideoStreams({})
    for(const stream of result.streams){
        console.log(`${stream.identity}} - ${stream.trackSid}`)
        if (matchAddress(stream.identity))
            VideoPlayer.createOrReplace(entity, { src: stream.trackSid, playing: true })
    }
  })
```
*adds VideoPlayer support for livekit video streaming 

## How to test the changes?

1. Launch the explorer
2. ...

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 08ae953</samp>

This pull request adds support for livekit video streams in the video player component and integrates the new comms API service that allows scenes to query the active video streams of other users. It updates the browser-interface package to use custom CDN versions of the `@dcl/protocol` and `@dcl/scene-runtime` packages that implement the comms API features. It also modifies the `IVideoPluginWrapper` interface and its implementations to handle different types of video sources, such as common, hls, or livekit. It fixes some syntax errors and code style issues in various files.

### CODE REVIEW

File: browser-interface/package.json [hunk 1]
```diff
@@ -62,9 +62,9 @@
     "@dcl/hashing": "^1.1.3",
     "@dcl/kernel-interface": "^2.0.0-20230512115658.commit-b582e05",
     "@dcl/legacy-ecs": "^6.11.11",
-    "@dcl/protocol": "^1.0.0-5623934099.commit-07e0626",
+    "@dcl/protocol": "https://sdk-team-cdn.decentraland.org/@dcl/protocol/branch//dcl-protocol-1.0.0-5661960573.commit-0a9d156.tgz",
     "@dcl/rpc": "^1.1.1",
-    "@dcl/scene-runtime": "7.0.6-20230721154039.commit-22b60b9",
+    "@dcl/scene-runtime": "https://sdk-team-cdn.decentraland.org/@dcl/scene-runtime/branch/feat/kernel-api-comms-api/dcl-scene-runtime-7.0.6-20230725213250.commit-3906c65.tgz",
```
Comment (pbosio) [inline browser-interface/package.json:67]: TODO: change

**Review by hugoArregui — APPROVED** [review]

File: browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts [hunk 2]
```diff
@@ -142,4 +142,8 @@ export class LivekitAdapter implements MinimumCommunicationsAdapter {
       data
     })
   }
+
+  getParticipants(): Map<string, RemoteParticipant>{
```
[the remark this answers is not in the document]
Comment (pbosio) [reply browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts]: changed here https://github.com/decentraland/unity-renderer/pull/5527/commits/6138fc97dafca3251380005bd8b73722e626d8b8

**Comment by aixaCode** [comment]
Is there a way for QA to test it?

**Comment by pbosio** [comment]
> Is there a way for QA to test it?

not sure, since owning a world or having permissions set to it in zone is a requirement for testing this

**Review by Kinerius — APPROVED** [review]

File: browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ export class LivekitAdapter implements MinimumCommunicationsAdapter {
   private voiceHandler: VoiceHandler
 
   constructor(private config: LivekitConfig) {
-    this.room = new Room()
+    this.room = new Room();
```
Comment (gonpombo8) [inline browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts:37]:
```suggestion
    this.room = new Room()
```

File: browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts [hunk 4]
```diff
@@ -142,4 +142,23 @@ export class LivekitAdapter implements MinimumCommunicationsAdapter {
       data
     })
   }
+
+  getActiveVideoStreams(): Map<string, ActiveVideoStreams> {
+    const result = new Map<string, ActiveVideoStreams>()
+    const participants = this.room.participants
+
+    for (let [sid, participant] of participants) {
```
Comment (gonpombo8) [inline browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts:150]:
```suggestion
    for (const [sid, participant] of participants) {
```

File: browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts [hunk 5]
```diff
@@ -142,4 +142,23 @@ export class LivekitAdapter implements MinimumCommunicationsAdapter {
...
+      if (participant.videoTracks.size > 0) {
+        const participantTracks = new Map<string, MediaStream>()
+        for (let [videoSid, track] of participant.videoTracks) {
```
Comment (gonpombo8) [inline browser-interface/packages/shared/comms/adapters/LivekitAdapter.ts:153]:
```suggestion
        for (const [videoSid, track] of participant.videoTracks) {
```

File: browser-interface/packages/unity-interface/loader.ts [hunk 6]
```diff
@@ -128,6 +137,24 @@ async function initializeWebRenderer(options: RendererOptions): Promise<Decentra
     // This function is called from the unity renderer to send messages back to the scenes
     BinaryMessageFromEngine(data: Uint8Array) {
       if (!!onBinaryMessage) onBinaryMessage(data)
+    },
+
+    RequestFromEngine(request: { type: string, payload: any }) {
+      const requestType = request.type as EngineRequestsNames
+      return onRequest(requestType, request.payload)
+
+      function onRequest<T extends EngineRequestsNames>(type: T, payload: IEngineRequests[T]) {
```
Comment (gonpombo8) [inline browser-interface/packages/unity-interface/loader.ts:146]: why we have a fn here instead of running the code ?

Comment (pbosio) [reply browser-interface/packages/unity-interface/loader.ts]: no reason at all, I was testing something al left the code like that :P
changed to remove that function

**Review by gonpombo8 — APPROVED** [review]

**Pull request merged**